24.1 Accessibility issues in demos fixes

dxArtemiusz opened the pull request:
The following demos have received changes to fix accessibility issues:
1) DataGrid - Row Selection
2) Localization - Using Intl
3) DataGrid - Filtering
4) FilterBuilder - With List
5) DataGrid - Master-Detail API
6) DataGrid - PDF Cell Customization
7) DataGrid - Custom New Record Position
8) DataGrid - Row Template
9) DataGrid - Row 3-rd Party Engine Template
10) DataGrid - Column Template
11) DataGrid - Column 3-rd Party Engine Template

Also, these demos were removed from exception list.

dxArtemiusz committed:
Using Intl Demo Fix

dxArtemiusz committed:
Filtering Demo Fix

Added aria-label to Sale Amount editor

dxArtemiusz committed:
With List Demo Fix

Added alt text to img element

dxArtemiusz committed:
Row Selection Demo Fix

Added alt text for image element.

Added visibility hidden for initial img element in jQuery and add logic to remove this style to display it on Row Selection as in other frameworks.

dxArtemiusz committed:
Master-Detail API Demo Fix

Added alt text for img element

dxArtemiusz committed:
PDF Cell Customization Demo Fix

Removed empty caption => DataGrid places accessible caption based on the dataField value

dxArtemiusz committed:
Custom New Record Position Demo Fix

Added text to add button. Text property is going as title attribute

dxArtemiusz committed:
Row Template Demo Fix

Added role to td and tr elements.

Added tabindex 0 and alt text to img element.

dxArtemiusz committed:
Row 3rd-Party Engine Template Demo Fix

Added role to td and tr elements.

Added tabindex 0 and alt text to img element.

dxArtemiusz committed:
Column Template and Column 3rd-Party Engine Template Demo Fix

Added alt to img elements

dxArtemiusz committed:
Updated exception demo list

Removed demos, that are fixed in this branch

dxArtemiusz committed:
Fix Lint

dxArtemiusz committed:
Fix mark-up

vladaskorohodova commented on apps/demos/Demos/DataGrid/Column3RdPartyEngineTemplate/jQuery/index.html:
```diff
@@ -18,7 +18,7 @@
       <script id="gridPhoto" type="text/html">
         <div class="photo">
-          <img src="{{ value }}" />
+          <img src="{{ value }}" alt="Image of {{ data.FirstName }} {{ data.LastName }}" />
```
```suggestion
          <img src="{{ value }}" alt="Picture of {{ data.FirstName }} {{ data.LastName }}" />
```

There's not much difference in "picture" and "image" but in this context picture suits better (since it's a photo and we're not talking about technical aspects).

vladaskorohodova commented on apps/demos/Demos/DataGrid/ColumnTemplate/Angular/app/app.component.html:
```diff
@@ -19,6 +19,9 @@
-    <img [src]="data.value" />
+    <img
+      [src]="data.value"
+      alt="Image of {{ data.data.FirstName }} {{ data.data.LastName }}"
```
```suggestion
      alt="Picture of {{ data.data.FirstName }} {{ data.data.LastName }}"
```

vladaskorohodova commented on apps/demos/Demos/DataGrid/ColumnTemplate/React/App.tsx:
```diff
@@ -3,7 +3,7 @@ import DataGrid, { Column, DataGridTypes } from 'devextreme-react/data-grid';
 import { employees } from './data.ts';
 
-const cellRender = (data: DataGridTypes.ColumnCellTemplateData) => (<img src={data.value} />);
+const cellRender = (data: DataGridTypes.ColumnCellTemplateData) => (<img src={data.value} alt={`Image of ${data.data.FirstName} ${data.data.LastName}`}/>);
```
```suggestion
const cellRender = (data: DataGridTypes.ColumnCellTemplateData) => (<img src={data.value} alt={`Picture of ${data.data.FirstName} ${data.data.LastName}`}/>);
```

vladaskorohodova commented on apps/demos/Demos/DataGrid/ColumnTemplate/ReactJs/App.js:
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import { employees } from './data.js';
 
-const cellRender = (data) => <img src={data.value} />;
+const cellRender = (data) => <img src={data.value} alt={`Image of ${data.data.FirstName} ${data.data.LastName}`}/>;
```
```suggestion
const cellRender = (data) => <img src={data.value} alt={`Picture of ${data.data.FirstName} ${data.data.LastName}`}/>;
```

vladaskorohodova commented on apps/demos/Demos/DataGrid/ColumnTemplate/Vue/App.vue:
```diff
@@ -29,7 +29,7 @@
       />
       <template #cellTemplate="{ data }">
-        <img :src="data.value">
+        <img :src="data.value" :alt="`Image of ${data.data.FirstName} ${data.data.LastName}`">
```
```suggestion
        <img :src="data.value" :alt="`Picture of ${data.data.FirstName} ${data.data.LastName}`">
```

vladaskorohodova commented on apps/demos/Demos/DataGrid/ColumnTemplate/jQuery/index.js:
```diff
@@ -10,7 +10,7 @@ $(() => {
       cellTemplate(container, options) {
         $('<div>')
-          .append($('<img>', { src: options.value }))
+          .append($('<img>', { src: options.value, alt: `Image of ${options.data.FirstName} ${options.data.LastName}` }))
```
```suggestion
          .append($('<img>', { src: options.value, alt: `Picture of ${options.data.FirstName} ${options.data.LastName}` }))
```

vladaskorohodova reviewed (commented):
I haven't finished the whole PR review yet so these are just the things I found so far

dxArtemiusz committed:
ExcelJS Cell Customization Demo Fixes

Removed caption

dxArtemiusz committed:
Apply suggestions from code review

Co-authored-by: Vlada Skorohodova <94827090+vladaskorohodova@users.noreply.github.com>
Signed-off-by: dxArtemiusz <111366900+dxArtemiusz@users.noreply.github.com>

dxArtemiusz committed:
FilterPanel demo fix

Added aria-label.

Replaced editorOptions from the onIntialized event handler to mark-up declaration in Angular

dxArtemiusz committed:
Excel Image Export Demo Fixes

Added alt text

dxArtemiusz committed:
Filtering API Demo Fixes

Added alt text

dxArtemiusz committed:
PDF Image Export Demo FIxes

Added alt text

dxArtemiusz committed:
Fix test excemptions

dxArtemiusz committed:
Merge branch '24_1_Accessibility_Issues_in_Demos_Fixes' of https://github.com/DevExpress/DevExtreme into 24_1_Accessibility_Issues_in_Demos_Fixes

dxArtemiusz committed:
Merge branch '24_1' into 24_1_Accessibility_Issues_in_Demos_Fixes

Signed-off-by: dxArtemiusz <111366900+dxArtemiusz@users.noreply.github.com>

dxArtemiusz committed:
Fix lint

dxArtemiusz committed:
Merge branch '24_1_Accessibility_Issues_in_Demos_Fixes' of https://github.com/DevExpress/DevExtreme into 24_1_Accessibility_Issues_in_Demos_Fixes

dxArtemiusz committed:
Fix ReactJS code style

dxArtemiusz committed:
Fix ReactJS 2

Raushen commented:
![](https://github.trello.services/images/mini-trello-icon.png) [♿️ Accessibility: Fix checker demo issues (2 day)](https://trello.com/c/gqLKRnCH/1191-%E2%99%BF%EF%B8%8F-accessibility-fix-checker-demo-issues-2-day)

dxArtemiusz committed:
Merge branch '24_1' into 24_1_Accessibility_Issues_in_Demos_Fixes

Signed-off-by: dxArtemiusz <111366900+dxArtemiusz@users.noreply.github.com>

dxArtemiusz committed:
Text change

dxArtemiusz committed:
Title Case Remove

vladaskorohodova commented on apps/demos/Demos/DataGrid/FilteringAPI/Angular/app/app.component.html:
```diff
@@ -1,7 +1,7 @@
-    <img src="../../../../images/logo-devav.png" />
-    <img src="../../../../images/logo-tasks.png" />
+    <img src="../../../../images/logo-devav.png" alt="DEVAV" />
+    <img src="../../../../images/logo-tasks.png" alt="tasks" />
```
```suggestion
    <img src="../../../../images/logo-tasks.png" alt="Tasks" />
```

vladaskorohodova commented on apps/demos/Demos/DataGrid/FilteringAPI/React/App.tsx:
```diff
@@ -43,9 +43,9 @@ const App = () => {
+          <img src="../../../../images/logo-devav.png" alt="DEVAV"/>
           &nbsp;
-          <img src="../../../../images/logo-tasks.png" />
+          <img src="../../../../images/logo-tasks.png" alt="tasks"/>
```
```suggestion
          <img src="../../../../images/logo-tasks.png" alt="Tasks"/>
```

vladaskorohodova commented on apps/demos/Demos/DataGrid/FilteringAPI/ReactJs/App.js:
```diff
@@ -37,9 +37,15 @@ const App = () => {
-          <img src="../../../../images/logo-tasks.png" />
+          <img
+            src="../../../../images/logo-tasks.png"
+            alt="tasks"
```
```suggestion
            alt="Tasks"
```

vladaskorohodova commented on apps/demos/Demos/DataGrid/FilteringAPI/Vue/App.vue:
```diff
@@ -2,8 +2,8 @@
-        <img :src="'../../../../images/logo-devav.png'">
-        <img :src="'../../../../images/logo-tasks.png'">
+        <img :src="'../../../../images/logo-devav.png'" alt="DEVAV">
+        <img :src="'../../../../images/logo-tasks.png'" alt="tasks">
```
```suggestion
        <img :src="'../../../../images/logo-tasks.png'" alt="Tasks">
```

vladaskorohodova commented on apps/demos/Demos/DataGrid/FilteringAPI/jQuery/index.html:
```diff
@@ -16,8 +16,8 @@
-          <img src="../../../../images/logo-devav.png" />
-          <img src="../../../../images/logo-tasks.png" />
+          <img src="../../../../images/logo-devav.png" alt="DEVAV" />
+          <img src="../../../../images/logo-tasks.png" alt="tasks" />
```
```suggestion
          <img src="../../../../images/logo-tasks.png" alt="Tasks" />
```

dxArtemiusz committed:
Update apps/demos/Demos/DataGrid/FilteringAPI/Angular/app/app.component.html

Co-authored-by: Vlada Skorohodova <94827090+vladaskorohodova@users.noreply.github.com>
Signed-off-by: dxArtemiusz <111366900+dxArtemiusz@users.noreply.github.com>

dxArtemiusz committed:
Update apps/demos/Demos/DataGrid/FilteringAPI/React/App.tsx

Co-authored-by: Vlada Skorohodova <94827090+vladaskorohodova@users.noreply.github.com>
Signed-off-by: dxArtemiusz <111366900+dxArtemiusz@users.noreply.github.com>

dxArtemiusz committed:
Update apps/demos/Demos/DataGrid/FilteringAPI/ReactJs/App.js

Co-authored-by: Vlada Skorohodova <94827090+vladaskorohodova@users.noreply.github.com>
Signed-off-by: dxArtemiusz <111366900+dxArtemiusz@users.noreply.github.com>

dxArtemiusz committed:
Update apps/demos/Demos/DataGrid/FilteringAPI/Vue/App.vue

Co-authored-by: Vlada Skorohodova <94827090+vladaskorohodova@users.noreply.github.com>
Signed-off-by: dxArtemiusz <111366900+dxArtemiusz@users.noreply.github.com>

dxArtemiusz committed:
Update apps/demos/Demos/DataGrid/FilteringAPI/jQuery/index.html

Co-authored-by: Vlada Skorohodova <94827090+vladaskorohodova@users.noreply.github.com>
Signed-off-by: dxArtemiusz <111366900+dxArtemiusz@users.noreply.github.com>
